docs: add steps to pre-install JetBrains IDE backend

EdwardAngert opened the pull request:
closes #13207 

[preview](https://coder.com/docs/@13207-preinstall-jetbrains/user-guides/workspace-access/jetbrains)

EdwardAngert committed:
copy and md

EdwardAngert committed:
Merge remote-tracking branch 'origin/main' into 13207-preinstall-jetbrains

EdwardAngert committed:
jetbrains initial copy and md edit

EdwardAngert committed:
Merge remote-tracking branch 'origin/main' into 13207-preinstall-jetbrains

* origin/main: (31 commits)
  docs: document issue with direnv and nix-shell on macOS (#15990) (#16015)
  chore: add spinner component (#16014)
  chore: bump react-syntax-highlighter from 15.5.0 to 15.6.1 in /site (#16003)
  chore: bump @radix-ui/react-dialog from 1.1.2 to 1.1.4 in /site (#15999)
  chore(provisionerd): close completeChan exactly once (#16011)
  fix(examples): add keep_locally to docker_image resources (#16012)
  feat: notify on workspace update (#15979)
  chore(apiversion): add guidelines regarding bumping apiversion (#16009)
  docs: add note about TLD and wildcards (#16008)
  chore: bump @storybook/preview-api from 8.4.6 to 8.4.7 in /site (#16002)
  ci: make sure golden files are actually up to date (#15989)
  chore: bump eslint-config-next from 14.2.18 to 14.2.22 in /offlinedocs (#15996)
  chore: bump @types/node from 20.17.6 to 20.17.11 in /site (#16001)
  chore: bump @types/node from 20.17.9 to 20.17.11 in /offlinedocs (#15995)
  chore: bump next from 14.2.16 to 14.2.22 in /offlinedocs (#15993)
  chore: bump the mui group across 1 directory with 6 updates (#15991)
  fix(enterprise/cli): add missing defer close of closer funcs (#15986)
  fix(cli/cliui): handle typed nil and null time in tables (#15984)
  test: fix incorrectly placed cleanup defer in dbtestutil (#15987)
  chore(scripts): remove unused ci-report (#15988)
  ...

EdwardAngert committed:
make fmt

EdwardAngert committed:
move JetBrains offline steps to separate doc; update links

EdwardAngert committed:
Merge remote-tracking branch 'origin/main' into 13207-preinstall-jetbrains

matifali commented on docs/user-guides/workspace-access/jetbrains/index.md:
thoughts about extracting Jetbrains Fleet to a separate page too?

EdwardAngert replied:
maybe - I'm still working through how this should flow and where Fleet fits in

just wanted to push this up and test the linting 😁

matifali replied:
I want us to have a similar page for zed.dev editor it would be similar to Fleet instructions.

EdwardAngert committed:
fix relative links

EdwardAngert committed:
s/offline/airgapped

EdwardAngert committed:
copy edit

EdwardAngert committed:
Merge remote-tracking branch 'origin/main' into 13207-preinstall-jetbrains

EdwardAngert committed:
copy edit

EdwardAngert committed:
add jetbrains-pre-install

EdwardAngert commented:
@matifali , I keep getting stuck on which parts need to be pre-installed, so I left it a little barebones in [jetbrains-pre-install.md](https://github.com/coder/coder/pull/15962/files#diff-491565c201376b77613020d477985502f635f54fa6163cd5530a0cc969de8b16)

Is there anything else that can be added to that doc that would help those steps?

matifali commented:
/cherry-pick release/2.18

matifali commented on docs/user-guides/workspace-access/jetbrains/jetbrains-pre-install.md:
````diff
@@ -0,0 +1,29 @@
+
+```shell
+mkdir ~/backends
+./jetbrains-clients-downloader-linux-x86_64-1867/bin/jetbrains-clients-downloader --products-filter <product-code> --build-filter <build-number> --platforms-filter linux-x64 --download-backends ~/backends
````
Can we give an example here?
Also, have you been able to verify if these instructions work as expected?

EdwardAngert replied:
@matifali : they don't - on its own, it's a page we're calling `Pre-install JetBrains Gateway in a template` but to me, that means that I should edit the template. when I add the lines in this doc to Dockerfile, I get:

```shell
Error: process "/bin/sh -c tar -xzvf jetbrains-clients-downloader-linux-x86_64-1867.tar.gz" did not complete successfully: exit code: 2

on main.tf line 89, in resource "docker_image" "main":
  89: resource "docker_image" "main" {
```

so I think we need to add more information about where these go and how they get there

---

> @matifali , I keep getting stuck on which parts need to be pre-installed, so I left it a little barebones in [jetbrains-pre-install.md](https://github.com/coder/coder/pull/15962/files#diff-491565c201376b77613020d477985502f635f54fa6163cd5530a0cc969de8b16)
> 
> Is there anything else that can be added to that doc that would help those steps?

matifali replied:
Yes. We should make sure the instructions we document work before we publish them.
If you can not make this work, you can try reaching @ericpaulsen or someone from their team, as they might have experience in trying this out.

matifali replied:
We can pair on this next week.

matifali committed:
Merge branch 'main' into 13207-preinstall-jetbrains

EdwardAngert commented:
leaving closed while we wait for updates/changes to JetBrains toolbox cc @matifali

matifali committed:
Improve JetBrains Gateway installation documentation

The subject line alone doesn't sufficiently explain the changes, so here's a body:

Enhance setup instructions with specific examples, cleanup commands,
and add Terraform configuration for the jetbrains-gateway module.
Update directory paths to be more user-friendly and consistent.

matifali committed:
Update JetBrains Gateway backend setup instructions

The command should be in startup_script instead of manual execution

matifali commented:
I went through the instructions again and was able to make it work. Updated the instructions. 

@EdwardAngert can you help rebase the PR branch and resolve conflicts. 

Next I am going to test this in a template.

EdwardAngert committed:
Merge remote-tracking branch 'origin/main' into 13207-preinstall-jetbrains

EdwardAngert committed:
align with changes from main

matifali committed:
Merge branch 'main' into 13207-preinstall-jetbrains

matifali committed:
Simplify JetBrains pre-installation guide

- Remove detailed script instructions
- Direct users to admin for backend setup
- Add Dockerfile example for pre-installing backend
- Organize existing content for clarity

matifali committed:
Simplify JetBrains pre-installation guide

Reduce indentation for the startup script block in the JetBrains guide to improve readability and consistency with other code examples.

matifali commented:
HI @EdwardAngert, It's all done from my side, but I think we should reorganize the pages to be in a different admin section.

matifali committed:
Merge branch 'main' into 13207-preinstall-jetbrains

EdwardAngert committed:
Merge branch 'main' into 13207-preinstall-jetbrains

EdwardAngert committed:
Merge branch 'main' into 13207-preinstall-jetbrains

EdwardAngert committed:
move gateway and update pre-install
